src/api/app/models/user.rb

```diff
@@ -358,8 +309,17 @@ def state=(value)
 
   # This method checks whether the given value equals the password when
   # hashed with this user's password hash type. Returns a boolean.
-  def password_equals?(value)
-    hash_string(value) == password
+  def password_old_equals?(value)
+    hash_string(value) == password_old
+  end
+
+  def convert_old_password(old_password)
+    self.password = old_password
```

Comment:
Don't we need to assign also the **passsword_confirmation** in the model?

Reply:
It's not needed here.

Reply:
```update(password_old: nil, password_old_hash_type: nil, password_old_salt: nil```